Evolutionary_loop/Loop_Parts/Part_B/part_b_job2.py

```diff
@@ -0,0 +1,188 @@
+#*******************************************************************************
+# File: part_b_job2.py
+#    Based on Part_B_VPol_job2.sh
+#
+#  Programmer: Jason Yao (yao.966@osu.edu)
+#
+#  Revision history:
+#     07/27/23  Jason Yao, Original version
+#
+#  Notes:
+#       * vertical ruler at column 80
+#  TODO:
+#       1 MOVE SECTION is not great; clean this up
+#       2. p0 & p1 need to be one process, and p0 & p2 need to be another.
+#          (i.e. load xfdtd needs to be absorbed)
+#
+#*******************************************************************************
+'''
+This is Part B2 of the loop, which prepares and runs output.xmacro with the
+relevant parameters (antenna type, population number, grid size, etc.)
+Output.xmacro writes the XFdtd simulation data to output files (.uan)
+The xmacro is mainly made of a xmacro-skeleton text file (see CAT SECTION).
+
+Usage: python3 part_b_job2.py <indiv> <gen> <npop> <working_dir> ... <nsections>
+
+Outout: XmacrosDir/simulation_PEC.xmacro
+
+For more information on the arguments: 
+  >> python3 part_b_job2.py -h
+'''
+
+
+import argparse
+import os
+import subprocess as sp
+from pathlib import Path
+from time import sleep
+
+
+'''
+This function counts the number of files that don't start with "." (hidden) 
+inside a folder "where". Subrirectories are not counted.
+Input: where (Path object)
+Output: counts (int)
+'''
+def count_flags(where=Path):
+    dummy = sp.run(f'cd {where}; ls | wc -l',
+                   shell=True, capture_output=True, text=True)
+    return int(dummy.stdout)
+
+
+def main(indiv, gen, npop, working_dir, run_name, xmacros_dir, xf_proj, 
+         geo_factor, num_keys, curved, nsections):
+## FLAG COUNT    
+
+    # Count the number of flags that indicate succesful simulations
+    flagpole = f'{working_dir}/Run_Outputs/{run_name}/GPUFlags'
+    num_flags = count_flags(flagpole)
+    # Sleep for a bit before counting again. Count until all jobs are finished.
+    while num_flags < npop:
+        sleep(60)
+        num_flags = count_flags(flagpole)
+    
+    # Upon XF job completion for all individuals, remove all flags (to prepare
+    # the directory for counting again for the next generation)
+    sp.run(f'rm {flagpole}/*', shell=True) # shell=True for wildcard ("*")
```

Comment:
```suggestion
   shutil.rmtree(flagpole)
```
You'll need to import `shutil` for this.

Reply:
Seems like this removes the directory `GPUFlags` (`flagpole`) as well, is there a pythonic way to use wildcards to delete all files instead of deleting the directory?

Reply:
Try this:
```suggestion
    for path in flagpole.glob('*'):
        path.unlink()
```
Is it only files or dirs too?